### PR TITLE
Make stub string key `recipe` go with values with more versatile names

### PR DESCRIPTION
**Pull request by dbogunowicz**
Currently:
`Model._validate_params()` will throw an error if `Model` receives any other recipe string arguments other than `?recipe=original` or `?recipe=transfer_learn`.

After the patch:
We account for the fact that we may be receiving arguments with more versatile names, such as `?recipe=transfer-text_classification` or `?recipe=original_fizz_buzz`

### CODE REVIEW

File: src/sparsezoo/model/model.py [hunk 1]
```diff
@@ -700,9 +700,17 @@ def _validate_params(params: Dict[str, str]):
...
                     f"expected set of string arguments {list(PARAM_DICT.keys())}!"
                 )
-            if PARAM_DICT[key] and value not in PARAM_DICT[key]:
-                raise ValueError(
-                    f"String argument {key} has value {value}, "
-                    "cannot be found in the "
-                    f"expected set of values {list(PARAM_DICT[key])}!"
-                )
+            if key == "recipe":
+                if not any(identifier in value for identifier in PARAM_DICT[key]):
```
Comment (dbogunowicz) [inline src/sparsezoo/model/model.py:704]: @bfineran do we have a convention for naming recipes? i.e
 every transfer learning recipe will be named `recipe_transfer_{...}` and every default recipe will be named `recipe_original_{...}`? If yes, we can change `identifier in value` to `value.startswith(identified)`.

Comment (bfineran) [reply src/sparsezoo/model/model.py:704]: `recipe_` is not present in any of these names right now (might be in update). `transfer` should be present in general, but this is not a hard rule. I think this hard check is fine for now. better than what we had before. can relax further as needed when new recipes are added

**Review by bfineran — DISMISSED** [review]

File: src/sparsezoo/model/model.py (hunk 1 above)
Comment (markurtz) [reply src/sparsezoo/model/model.py:704]: Standards moving forward will be renaming the base recipe to recipe.md and we will still need to support recipe_original for backwards compatibility. recipe_{name} is the future general format and recipe_transfer is a special match for transfer types that the user can request for use

Comment (markurtz) [reply src/sparsezoo/model/model.py:704]: We should remove ALLOWED_RECIPE_VALUES and respect anything that is listed from the API. Big thing is making sure that when people request a transfer checkpoint that we don't require the full transfer name and can assume which one to return. If there are multiple there, then we fail

**Review by markurtz — CHANGES_REQUESTED** [review]
Comments left in convo

**Review by bfineran — APPROVED** [review]
approving - confirming new logic is for stubs with a `recipe` param, we will return the recipe that matches `recipe_{recipe}`, if more than one match then we raise an exception

**Review by markurtz — APPROVED** [review]
Approved, need to do some regression testing for older recipe flows to make sure those are working with this

**Pull request merged**